feature extraction

mihirneal opened the pull request:
pushing scripts for stimuli feature extraction + updated README

mihirneal committed:
initial feature extractions

Commit:
bug fixes

mihirneal committed:
feature_extraction + updated README

clane9 commented on README.md:
```diff
@@ -1,15 +1,42 @@
+# Predicting Brain Responses to Natural Movies with Multimodal LLMs
 
-Repository for MedARC group tackling the [Algonauts 2025 challenge](https://algonautsproject.com/).
+**Cesar Kadir Torrico Villanueva¹⁴\*, Jiaxin Cindy Tu¹²\*, Mihir Tripathy¹³\*, Connor Lane¹⁴\*, Rishab Iyer¹⁵, Paul S. Scotti¹⁴**
```
check cesar's affiliation, if you decide to keep the author list.

clane9 commented on README.md:
```diff
@@ -1,15 +1,42 @@
-# MedARC Algonauts 2025 submission
+# Predicting Brain Responses to Natural Movies with Multimodal LLMs
```
imo the author list a little formal for github. I like better just having the link to arxiv at the top, and then the bibtex block at the bottom for citation.

mihirneal replied:
i'll just remove it

clane9 commented on README.md:
```diff
@@ -1,15 +1,42 @@
+
+### Main Environment
+
+For the main application, run the following command. This will create a default `.venv` and install the project's primary dependencies from the lock file.
```
I think should still include the link to `uv`, since people might not have it/know what it is.

clane9 commented on README.md:
```diff
@@ -1,15 +1,42 @@
+
+If you need to run the feature extraction scripts, they require a separate environment with different dependencies.
+
+1.  **Create the environment** named `ext`:
```
it would be cleaner to have a single environment. I don't think I had any strict version requirements, so should be possible to merge them. if we find any version issues later we can fix them. though it's unlikely, since we're not continuing to work on this code really.

mihirneal replied:
it would be, yes, but flash attention keeps on breaking with different torch versions and dependencies. i'll try to combine them

mihirneal replied:
yep, flash_attention is breaking with the main env. there might be some rogue package that's causing this (probably the version on torch). i think it's just cleaner to have separate environments

clane9 commented on feature_extraction/internvl3_video_features.py:
```diff
@@ -0,0 +1,1349 @@
+
+
+
+class HuggingFaceFeatureExtractor:
```
ideally would have a shared utility for this and not copy into every script, since it's generic. could you use the version we have in `feature_extractor.py`?

clane9 reviewed (commented):
Thanks @mihirneal. I left some comments. Biggest one is I think it would be nice to have a single unified environment, rather than one for features and one for model training. What do you think? I can merge the  environments if you agree.

clane9 committed:
merge feature extraction environment with main env

- upgraded torch to 2.7.1 to fix an incompatibility with flash attention
- also added citation to readme

clane9 committed:
actually, can install flash-attn the usual way now

since we upgraded torch, can just install flash-attn the usual way from
pypi (I think).

mihirneal committed:
merged feature extraction class

clane9 committed:
exclude feature extraction scripts from ruff

clane9 committed:
update `FeatureExtractor`

clane9 committed:
update feature extraction scripts

- minor changes to feature extractor api
- fix trailing whitespace (but don't run ruff)
